nil/getting-started/essentials/receiving-ext-messages.mdx

```diff
@@ -0,0 +1,41 @@
+# Receiving external messages
+
+This tutorial explains how exactly a contract can accept external messages.
+
+## `verifyExternal()`
+
+The contract receiving [**an external message**](../../core-concepts/shards-parallel-execution##internal-vs-external-messages) must implement an additional function so that this call is processed. Unless this function is present, a contract cannot accept external messages. The function signature:
+
+```solidity showLineNumbers
+function verifyExternal(
+        uint256 seqno, 
+        bytes calldata authData 
+    ) external view returns (bool)
+```
+
+The purpose of the function is to limit who can call the receiver contract via an external message. Its body can hold any logic for checking the message signature.
+
+As the receiving contract is charged with paying for external messages, the `verifyExternal()` function is needed so that external parties do not accidentally (or maliciously) drain the balance of the receiving contract.
+
+:::info
+
+`verifyExternal()` is executed every time a contract is called via an external message.
+
+:::
+
+## Example
+
+This example contains a simple mechanism for verifying `authData`:
+
+```solidity showLineNumbers
+contract Receiver {
+    bytes private validAuthData = AUTH_DATA;
+
+    function verifyExternal(uint256 seqno, bytes authData) external view returns (bool) {
+        if (keccak256(authData) == keccak256(validAuthData) && callData.length > 0) {
```

Comment:
* `callData` is not defined.
* Why are we comparing keccaks instead of comparing bytes directly?
* An attacker can copy validAuthData from the state of the contract and use it to satisfy the check.